app/scripts/directives/overview/serviceInstanceRow.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+angular.module('openshiftConsole').component('serviceInstanceRow', {
+  controller: [
+    '$filter',
+    'DataService',
+    'rowMethods',
+    '$uibModal',
+    ServiceInstanceRow
+  ],
+  controllerAs: 'row',
+  bindings: {
+    apiObject: '<',
+    state: '<'
+  },
+  templateUrl: 'views/overview/_service-instance-row.html'
+});
+
+function ServiceInstanceRow($filter, DataService, rowMethods, $uibModal) {
+  var row = this;
+  _.extend(row, rowMethods.ui);
+
+  // returns the best user-friendly name available for the instance
+  var getDisplayName = function() {
+    var serviceClassName = row.apiObject.spec.serviceClassName;
+    var instanceName = row.apiObject.metadata.name;
+    var serviceClassDisplayName = _.get(row, ['state','serviceClasses', serviceClassName, 'OSBMetadata', 'displayName']);
+    return serviceClassDisplayName || serviceClassName || instanceName;
+  };
+
+  var getBindings = function() {
+    return _.filter(row.state.bindings, function(binding) {
+      return binding.spec.instanceRef.name === row.apiObject.metadata.name;
+    });
+  };
+
+  row.$onChanges = function() {
+    row.notifications = rowMethods.getNotifications(row.apiObject, row.state);
+    row.displayName = getDisplayName();
+    row.instanceBindings = getBindings();
+  };
+
+  row.getSecretForBinding = function(binding) {
+    return binding && row.state.secrets[binding.spec.secretName];
+  }
+
+
+  row.deprovision = function() {
```

Comment:
@jwforres @ncameronbritt Q... so deprovision is pretty destructive.  Do we want to do the same thing that we do for deleting a project?  IE, type in the name of the instance before we let you go forward?  Far as I understand, there is no going back, so if you deprovision something like a database you might 😢 😭 .